packages/bigtable/test/instance.js

```diff
@@ -0,0 +1,687 @@
+        assert.deepEqual(grpcOpts, {
+          service: 'BigtableInstanceAdmin',
+          method: 'listClusters'
+        });
+
+        assert.strictEqual(reqOpts.parent, INSTANCE_ID);
+        done();
+      };
+
+      instance.getClusters(assert.ifError);
+    });
+
+    it('should copy all query options', function(done) {
+      var fakeOptions = {
+        a: 'a',
+        b: 'b'
+      };
+
+      instance.request = function(grpcOpts, reqOpts) {
+        Object.keys(fakeOptions).forEach(function(key) {
+          assert.strictEqual(reqOpts[key], fakeOptions[key]);
+        });
+
+        assert.notStrictEqual(reqOpts, fakeOptions);
+        done();
+      };
+
+      instance.getClusters(fakeOptions, assert.ifError);
+    });
+
+    it('should return an error to the callback', function(done) {
+      var error = new Error('err');
+      var response = {};
+
+      instance.request = function(grpcOpts, reqOpts, callback) {
+        callback(error, response);
+      };
+
+      instance.getClusters(function(err, clusters, nextQuery, apiResponse) {
+        assert.strictEqual(err, error);
+        assert.strictEqual(clusters, null);
+        assert.strictEqual(nextQuery, null);
+        assert.strictEqual(apiResponse, response);
+        done();
+      });
+    });
+
+    it('should return an array of cluster objects', function(done) {
+      var response = {
+        clusters: [{
+          name: 'a'
+        }, {
+          name: 'b'
+        }]
+      };
+
+      var fakeClusters = [
+        {},
+        {}
+      ];
+
+      instance.request = function(grpcOpts, reqOpts, callback) {
+        callback(null, response);
+      };
+
+      var clusterCount = 0;
+
+      instance.cluster = function(name) {
+        assert.strictEqual(name, response.clusters[clusterCount].name);
+        return fakeClusters[clusterCount++];
+      };
+
+      instance.getClusters(function(err, clusters, nextQuery, apiResponse) {
+        assert.ifError(err);
+        assert.strictEqual(clusters[0], fakeClusters[0]);
+        assert.strictEqual(clusters[1], fakeClusters[1]);
+        assert.strictEqual(nextQuery, null);
+        assert.strictEqual(apiResponse, response);
+        done();
+      });
+    });
+
+    it('should provide a nextQuery object', function(done) {
+      var response = {
+        clusters: [],
+        nextPageToken: 'a'
+      };
+
+      var options = {
+        a: 'b'
+      };
+
+      instance.request = function(grpcOpts, reqOpts, callback) {
+        callback(null, response);
+      };
+
+      var callback = function(err, clusters, nextQuery) {
+        var expectedQuery = extend({}, options, {
+          pageToken: response.nextPageToken
+        });
+
+        assert.ifError(err);
+        assert.deepEqual(nextQuery, expectedQuery);
+        done();
+      };
+
+      instance.getClusters(options, callback);
+    });
+  });
+
+  describe('getTables', function() {
+    var views = {
```

Comment:
We have to stub `Table.views` here, otherwise we don't know `getTables` is getting the correct data from the correct place.